### ISSUE TITLE
Supporting latest version of `@shopify/shopify-api` (version 6)

### ISSUE THREAD

**Issue by brendenpalmer** [TheSecurityDev/simple-koa-shopify-auth#14]
Are there plans to update `simple-koa-shopify-auth` to support the latest version of `@shopify/shopify-api` (version 6)? There have been quite a few breaking changes as part of this upgrade: https://github.com/Shopify/shopify-api-js/blob/main/docs/migrating-to-v6.md.

**Comment by TheSecurityDev** [comment]
Not in the immediate future unfortunately due to all the breaking changes. I may try if more people are interested, though.

**Comment by axmachina** [comment]
Indeed, it's a substantive amount of changes. I'm looking for ways to fix.

**Comment by matthewattanasio** [comment]
I would be very interested in this update!

**Comment by matthewattanasio** [comment]
From what I understand, Shopify will be depreciating Shopify API version 5 around October 2023. 

Is there going to be an update to support Shopify API 6+ or will I have move on the Nodejs / Koa / Next monolith

**Comment by TheSecurityDev** [comment]
> From what I understand, Shopify will be depreciating Shopify API version 5 around October 2023. 
> 
> Is there going to be an update to support Shopify API 6+ or will I have move on the Nodejs / Koa / Next monolith 

I'm fairly certain it will be updated before then. If not, just keep nagging me about it 😅. I may try to work on this soon.

**Comment by tolgap** [comment]
It would be great to see v6 and v7 supported. I've built our app back in 2019 using `shopify-koa-auth` and am in no way able to migrate away from Koa to Express.

**Comment by TheSecurityDev** [comment]
> From what I understand, Shopify will be depreciating Shopify API version 5 around October 2023.
> 
> Is there going to be an update to support Shopify API 6+ or will I have move on the Nodejs / Koa / Next monolith

I'm not sure exactly what you mean by it being deprecated. Are you referring to the API versions that are available, like 2023-04? Because you can specify any version as a string without using the ApiVersion enum from the library. Since this is a library, there's really no reason it should stop working unless something else changes.

**Comment by matthewattanasio** [comment]
Thank you for getting back to me @TheSecurityDev, so based on this resource https://shopify.dev/docs/api/usage/versioning, the way I understand it is our apps using your package can only go up to version **2022-10**, which means it will be supported until **2023-10**. 

However afterwards, if nothing changes, our apps will receive a depreciated notice for live apps and there stores wont be able to install and our app will be delisted on the shop store. https://shopify.dev/docs/api/usage/versioning#deprecation-practices

It would be amazing if this package could support the latest v7 of the shopify API ( not much has changed since v6, but v6 is a major change). because going from v5 to v6 is going to require a lot of work and testing on all of our behalfs due to amount of changes introduce, so its always better to do this earlier then later. 

Keen to hear your thoughts mate.

**Comment by TheSecurityDev** [comment]
> the way I understand it is our apps using your package can only go up to version 2022-10.

@matthewattanasio  Can you explain why this is the case? As I stated earlier, you don't have to rely on the ApiVersion constants that the @shopify/shopify-api library provides. You can use any string you want and specify the version yourself. I'm successfully using the latest version (2023-04) with my app. Unless there's some change in the library I don't know about, it should work still.

**Comment by matthewattanasio** [comment]
Very interesting to hear @TheSecurityDev I didnt know this was possible. 

So really, until Shopify introduces a change in the API that breaks "@shopify/shopify-api": "5.3.0" then we don't have to worry about updating to version 6 or higher?

**Comment by TheSecurityDev** [comment]
@matthewattanasio That's correct.

**Comment by brian-matuszak** [comment]
@TheSecurityDev Are there still any plans for `@shopify/shopify-api` v6+ support? Or has that been abandoned?

**Comment by TheSecurityDev** [comment]
> @TheSecurityDev Are there still any plans for `@shopify/shopify-api` v6+ support? Or has that been abandoned?

Sorry, it's just not really a priority for me :( I've been getting along with the old API and don't really plan to update right now. I would have to update my app in order to be able to make sure the update would work.
